### PR TITLE
Kogito 2528 : Kogito [BPMN] Reuse Data Types across the process

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin this is the initial PR, just sending to check sonar cloud results. I am missing writing several tests. Have tested both Kogito and BC to make sure the functionality is uniform. Since this PR depends on the work from Data Objects I did the work on top of the DataObjects branch I still have.

### CODE REVIEW

**Comment by inodeman** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/util/StringUtils.java [hunk 1]
```diff
@@ -112,6 +112,38 @@ public static String replaceIllegalCharsAttribute(final String value) {
...
+     * Url encoding is not supported on the Engine side so this method should be used for attribute values.
+     * @param value a string to escape illegal characters on the client side for Data Objects
+     * @return an escaped string
+     */
+    public static String replaceIllegalCharsForDataObjects(final String value) {
+        if (isEmpty(value)) {
+            return value;
+        }
+
+        final StringBuilder sb = new StringBuilder();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/util/StringUtils.java:128]: Use can use just Strings... no need for `StringBuilder` anymore

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/validation/impl/DiagramValidatorTest.java [hunk 2]
```diff
@@ -283,4 +357,52 @@ public void testValidateDiagramWithNullBean() {
         when(diagram.getGraph()).thenReturn(graphTestHandler.graph);
         tested.validate(diagram, this::assertNoErrors);
     }
+
+    public class ManagedInstanceStub<T> implements ManagedInstance<T> {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/validation/impl/DiagramValidatorTest.java:361]: You can use a class named `ManagedInstanceMock` or some already existing `ManagedInstanceStub`, I think there is no need for creating a new stubb class for this, already exist.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/BPMNCategories.java [hunk 3]
```diff
@@ -38,5 +38,4 @@
 
     public static final String SERVICE_TASKS = "ServiceTasks";
 
-    public static final String TEXT_ANNOTATION = "TextAnnotations";
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/BPMNCategories.java:41]: why this is removed? maybe are you mixing this with the Data Objects code?

**Comment by inodeman** [comment]
Closing PR, will create a new one without Data Objects Code

**Pull request closed without merging**